lib/common/cmdline.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <config.h>
+#include <glib.h>
+
+#include <crm/common/cmdline.h>
+#include <crm/common/util.h>
+
+GOptionContext *
+pcmk__build_arg_context(const char *param_string, const char *fmts,
+                        gboolean *version, gboolean *quiet, GOptionArgFunc verbose_cb,
+                        char **output_ty, char **output_dest) {
```

Comment:
An alternative to taking all the args would be to define a "common args" struct and pass that instead. That would make adding new members easier in the future.